horovod/spark/common/store.py

```diff
@@ -532,23 +533,37 @@ def matches(cls, path):
         return path.startswith(cls.FS_PREFIX)
 
 
+# If `_DBFS_PATH_MAPPING_TO_PATH` is not None, map `/dbfs/...` path to `{_DBFS_PATH_MAPPING_TO_PATH}/...`
+# This is used in testing.
+_DBFS_PATH_MAPPING_TO_PATH = None
+
+
 class DBFSLocalStore(FilesystemStore):
     """Uses Databricks File System (DBFS) local file APIs as a store of intermediate data and
     training artifacts.
 
     Initialized from a `prefix_path` starts with `/dbfs/...`, `file:///dbfs/...` or `dbfs:/...`, see
     https://docs.databricks.com/data/databricks-file-system.html#local-file-apis.
     """
+
+    DBFS_PATH_FORMAT_ERROR = "The provided path is not a DBFS path: {}, Please provide a path " \
+                             "starting with `/dbfs/...` or `dbfs:/...` or `file:/dbfs/...` or " \
+                             "`file:///dbfs/...`."
+
     def __init__(self, prefix_path, *args, **kwargs):
         prefix_path = self.normalize_path(prefix_path)
-        if not prefix_path.startswith("/dbfs/"):
-            warnings.warn("The provided prefix_path might be ephemeral: {} Please provide a "
-                          "`prefix_path` starting with `/dbfs/...`".format(prefix_path))
+        if not DBFSLocalStore.matches_dbfs(prefix_path):
+            raise ValueError(DBFSLocalStore.DBFS_PATH_FORMAT_ERROR.format(prefix_path))
         super(DBFSLocalStore, self).__init__(prefix_path, *args, **kwargs)
 
+    @staticmethod
+    def _get_dbfs_path_format_error(path):
+        return
+
     @classmethod
     def matches_dbfs(cls, path):
-        return path.startswith("dbfs:/") or path.startswith("/dbfs/") or path.startswith("file:///dbfs/")
+        return path.startswith("dbfs:/") or path.startswith("/dbfs/") or path.startswith("file:///dbfs/") \
+                or path.startswith("file:/dbfs/")
 
     @staticmethod
     def normalize_path(path):
```

Comment:
I think this static method is easy to test. Please add a unit test for the use cases of this method.